File: src/dataset/javascript/saddle-points_exercism-ThomasZumsteg/claude/ClaudeSonnet4_saddle-points_v3.js
var Matrix = function(matrix) {
	this.rows = matrix.split("\n").map(function(row) { 
		return row.replace(/^\s+|\s+$/g,'').split(' ')
			.map(function(e) { return parseInt(e, 10)});
	});

	const numRows = this.rows.length;
	const numCols = this.rows[0].length;
	
	this.columns = new Array(numCols);
	for (let col = 0; col < numCols; col++) {
		this.columns[col] = new Array(numRows);
		for (let row = 0; row < numRows; row++) {
			this.columns[col][row] = this.rows[row][col];
		}
	}
	
	this.saddlePoints = saddlePoints(this);
}

function saddlePoints(matrix) {
	const numRows = matrix.rows.length;
	const numCols = matrix.columns.length;
	
	const row_max = new Array(numRows);
	for (let i = 0; i < numRows; i++) {
		let max = matrix.rows[i][0];
		for (let j = 1; j < matrix.rows[i].length; j++) {
			if (matrix.rows[i][j] > max) {
				max = matrix.rows[i][j];
			}
		}
		row_max[i] = max;
	}
	
	const column_min = new Array(numCols);
	for (let i = 0; i < numCols; i++) {
		let min = matrix.columns[i][0];
		for (let j = 1; j < matrix.columns[i].length; j++) {
			if (matrix.columns[i][j] < min) {
				min = matrix.columns[i][j];
			}
		}
		column_min[i] = min;
	}
	
	const saddlePoints = [];
	for (let r = 0; r < numRows; r++) {
		for (let c = 0; c < numCols; c++) {
			const element = matrix.rows[r][c];
			if (element === row_max[r] && element === column_min[c]) {
				saddlePoints.push([r, c]);
			}
		}
	}
	
	return saddlePoints;
}

module.exports = Matrix;

File: src/dataset/python/saddle-points_exercism-ThomasZumsteg/claude/ClaudeSonnet4_saddle_points_v3.py
"""Finds saddle points in a rectangular 2d array"""

def saddle_points(matrix):
    """Finds points that are the largest in their row
    and smallest in their column
    """
    if not matrix or not matrix[0]:
        return set()
    
    rows = len(matrix)
    cols = len(matrix[0])
    
    for row in matrix:
        if len(row) != cols:
            raise ValueError
    
    row_maxes = [max(row) for row in matrix]
    col_mins = [min(matrix[r][c] for r in range(rows)) for c in range(cols)]
    
    saddles = set()
    for r in range(rows):
        for c in range(cols):
            if row_maxes[r] == col_mins[c]:
                saddles.add((r, c))
    
    return saddles